### PR TITLE
[CARBONDATA-30]Record load performance statistics

### PR DESCRIPTION
**Pull request by Zhangshunyu**
## What changes were proposed in this pull request?

1)We can use a parameter "enable.data.loading.statistics" which can be configured by user to determine whether the statistics will be recorded and calculated during data loading, the default value is false.
2)We defined a dummy util, if we don't need to record the statistics, it will do nothing.

For example, we set "enable.data.loading.statistics" to "true" and run CarbonExample, the message as following:

Data load request has been received for table default.t3

```
========== TIME_STATISTICS PartitionID: 0==========
STAGE 1 ->Load csv to DataFrame and generate block distinct values: 0.33(s)
STAGE 2 ->Global dict shuffle and write dict file: 0.106(s)
STAGE 2.1 ->  |_maximum distinct column shuffle time: 0.031(s)
STAGE 2.2 ->  |_maximum distinct column write dict file time: 0.053(s)
STAGE 3 ->LRU cache load: 0.001(s)
STAGE 4 ->Total cost of gen surrogate key, sort and write to temp files: 0.625(s)
STAGE 4.1 ->  |_read csv file: 0.47(s)
STAGE 4.2 ->  |_transform to surrogate key: 0.377(s)
STAGE 4.3 ->  |_sort rows and write to temp file: 0.366(s)
STAGE 5 ->Tansform to MDK, compress and write fact files: 0.165(s)
========== BLOCK_INFO ==========
BLOCK_INFO ->Node host: localhost
BLOCK_INFO ->The block count in this node: 1
===============Load_Speed_Info===============
Total Num of Records Processed: 1000
Total Time Cost: 1.227(s)
Total Load Speed: 814records/s
Generate Dictionaries Speed: 3030records/s
Read CSV Speed: 2127 records/s
Generate Surrogate Key Speed: 2652 records/s
Sort Key/Write Temp Files Speed: 2732 records/s
MDK Step Speed: 6060 records/s
=============================================
```
## How was this patch tested?

Pass the exist tests.

### CODE REVIEW

**Comment by Zhangshunyu** [comment]
@gvramana pls review

**Comment by ravipesala** [comment]
This PR has been reviewed and given comments at https://github.com/apache/incubator-carbondata/pull/8/files .

File: core/src/main/java/org/carbondata/core/util/CarbonLoadStatisticsDummy.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+  }
+
+  @Override
+  public void recordDicWriteFileMaxTime(Long dicWriteFileTimePart) {
+
+  }
+
+  @Override
+  public void recordSurrogatekeysTotalTime(String partitionID,
```
Comment (ravipesala) [inline core/src/main/java/org/carbondata/core/util/CarbonLoadStatisticsDummy.java]: Don't use surrogate key, use Dictionary

File: core/src/main/java/org/carbondata/core/util/CarbonLoadStatisticsImpl.java [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+      dicWriteFileCostTime = dicWriteFileTimePart;
+    }
+  }
+
+
+  public double getLruCacheLoadTime() {
+    return lruCacheLoadTime;
+  }
+
+  public void recordSurrogatekeysTotalTime(String partitionID,
```
Comment (ravipesala) [inline core/src/main/java/org/carbondata/core/util/CarbonLoadStatisticsImpl.java]: Don't use surrogate anywhere, use dictionary every where, even also in logs

**Comment by ravipesala** [comment]
LGTM except few comments. This PR requires refactoring, we can take refactoring in another PR.

**Comment by Zhangshunyu** [comment]
@ravipesala Done, please check.

**Comment by Zhangshunyu** [comment]
@ravipesala Done, please check.

**Comment by ravipesala** [comment]
LGTM

**Pull request merged**